app/code/Magento/Catalog/Model/Product/Copier.php

```diff
@@ -136,10 +138,13 @@ private function setStoresUrl(Product $product, Product $duplicate) : void
             }
             do {
                 $urlKey = $this->modifyUrl($urlKey);
-                $duplicate->setUrlKey($urlKey);
-                $duplicate->setData('url_path', null);
+                $productIds = [$duplicate->getId()];
```

Comment:
Hi @webkul-paul,

According to following code - `$duplicate->getId()` will always return null
https://github.com/magento/magento2/blob/a45d5196c23985068c8250a9d79fae25e89b4a1d/app/code/Magento/Catalog/Model/Product/Copier.php#L58-L88

Looks like it will not work, but have to check it additionally.

Reply:
Hi @ihor-sviziev 
After this line 
$this->setDefaultUrl($product, $duplicate); line number 80
I can get the duplicate product's id with $duplicate->getId()

Reply:
Ah, yes, you're correct.